traits/editor_factories.py

```diff
@@ -148,7 +148,11 @@ def _instance_handler_factory(handler):
     if isinstance(handler, TraitInstance):
         return handler.aClass
     elif isinstance(handler, BaseInstance):
-        return handler.default_value
+        result = handler.default_value
+        if isinstance(result, tuple):
+            return lambda: result[0](*result[1], **result[2])
```

Comment:
it'll improve the code readability if you could assign `result[0]`, `result[1]` and `result[2]` to meaningful variables e.g. `args` and `kwargs` for `result[1]` and `result[2]` if im not wrong and `default_value_getter` for `result[0]`